apps/web/src/components/MintLinkButton.tsx

```diff
@@ -0,0 +1,144 @@
+import { useCallback, useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { MintLinkButtonFragment$key } from '~/generated/MintLinkButtonFragment.graphql';
+import { FxHashLogoIcon } from '~/icons/FxHashLogoIcon';
+import { MintFunLogoIcon } from '~/icons/MintFunLogoIcon';
+import { ProhibitionLogoIcon } from '~/icons/ProhibitionLogoIcon';
+import { ZoraLogoIcon } from '~/icons/ZoraLogoIcon';
+import { contexts } from '~/shared/analytics/constants';
+import colors from '~/shared/theme/colors';
+import { MINT_LINK_DISABLED_CONTRACTS } from '~/shared/utils/communities';
+import {
+  getMintUrlWithReferrer,
+  MINT_LINK_CHAIN_ENABLED,
+} from '~/shared/utils/getMintUrlWithReferrer';
+
+import { Button, ButtonProps } from './core/Button/Button';
+import VerifyNavigationPopover from './core/GalleryLink/VerifyNavigationPopover';
+import { HStack } from './core/Spacer/Stack';
+
+type Props = {
+  tokenRef: MintLinkButtonFragment$key;
+  overwriteURL?: string;
+  referrerAddress?: string;
+  size?: 'sm' | 'md';
+} & ButtonProps;
+
+export function MintLinkButton({
+  overwriteURL,
+  referrerAddress,
+  tokenRef,
+  size = 'md',
+  variant = 'primary',
+  ...props
+}: Props) {
+  const token = useFragment(
+    graphql`
+      fragment MintLinkButtonFragment on Token {
+        definition {
+          community {
+            contract {
+              mintURL
+              contractAddress {
+                chain
+                address
+              }
+            }
+          }
+        }
+      }
+    `,
+    tokenRef
+  );
+
+  const { showModal } = useModalActions();
+
+  const tokenContractAddress =
+    token?.definition?.community?.contract?.contractAddress?.address ?? '';
+  const tokenChain = token?.definition?.community?.contract?.contractAddress?.chain ?? '';
+  const { url: mintURL, provider: mintProviderType } = getMintUrlWithReferrer(
+    overwriteURL ?? token?.definition?.community?.contract?.mintURL ?? '',
+    referrerAddress ?? ''
+  );
+
+  const mintProvider: {
+    buttonText: string;
+    icon: React.ReactNode;
+  } | null = useMemo(() => {
+    if (mintProviderType === 'Zora') {
+      return {
+        buttonText: 'mint on zora',
+        icon: <ZoraLogoIcon />,
+      };
+    } else if (mintProviderType === 'MintFun') {
+      return {
+        buttonText: 'mint on mint.fun',
+        icon: <MintFunLogoIcon />,
+      };
+    } else if (mintProviderType === 'FxHash') {
+      return {
+        buttonText: 'mint on fxhash',
+        icon: <FxHashLogoIcon mode={variant === 'primary' ? 'light' : 'dark'} />,
+      };
+    } else if (mintProviderType === 'Prohibition') {
+      return {
+        buttonText: 'mint on prohibition',
+        icon: <ProhibitionLogoIcon mode={variant === 'primary' ? 'light' : 'dark'} />,
```

Comment:
we should add `EnsembleIcon` as well, will provide more details on slack

Reply:
Added `Ensemble` to web and mobile!

![CleanShot 2023-12-13 at 15 33 55@2x](https://github.com/gallery-so/gallery/assets/4480258/e2337bd5-38c5-45ce-903a-9477775ea592)


![CleanShot 2023-12-13 at 15 33 29@2x](https://github.com/gallery-so/gallery/assets/4480258/b0039f46-c0dd-4eb5-9966-4abf68abd06e)

Reply:
![CleanShot 2023-12-13 at 15 39 37@2x](https://github.com/gallery-so/gallery/assets/4480258/8c241421-4285-4eff-9961-f3e677a0bc23)